src/Connections/GitlabRepo.ts

```diff
@@ -109,8 +128,9 @@ export class GitLabRepoConnection extends CommandConnection {
 
     private readonly debounceMRComments = new Map<string, {comments: number, author: string, timeout: NodeJS.Timeout}>();
 
-    public static async provisionConnection(roomId: string, requester: string, data: Record<string, unknown>, as: Appservice, tokenStore: UserTokenStore, instance: GitLabInstance, gitlabConfig: BridgeConfigGitLab) {
+    public static async provisionConnection(roomId: string, requester: string, data: Record<string, unknown>, as: Appservice, tokenStore: UserTokenStore, instanceName: string, gitlabConfig: BridgeConfigGitLab) {
         const validData = validateState(data);
+        const instance = gitlabConfig.instances[instanceName];
         const client = await tokenStore.getGitLabForUser(requester, instance.url);
```

Comment:
This is another object that's likely not guaranteeing to return an instance object, but could also return `undefined`.
The later would break the lookup of `.url` and throw an error.